src/TextFieldCatalog.js

```diff
@@ -126,29 +130,161 @@ const CharacterCounter = () => (
   <div className='mdc-text-field-character-counter'>0 / {TEXT_FIELD_MAX_LENGTH}</div>
 );
 
-const TextFieldCatalog = () => (
+const getTextFieldConfig = (props) => {
+  let iconsValue = [];
+  const {search} = props.location;
+  if (search) {
+    const searchObject = getUrlParamsFromSearch(search);
+    if (searchObject && searchObject.icons) {
+      iconsValue = searchObject.icons.split(',');
+    }
+  }
+
+  const TextFieldConfig = {
+    options: {
+      header: {
+        type: 'label',
+        name: 'Options',
+      },
+      type: {
+        type: 'select',
+        name: 'Variant',
+        urlParam: 'type',
+        value: 'filled', // default select first option
+        options: [
+          {
+            label: 'Filled',
+            value: 'filled',
+          },
+          {
+            label: 'Outlined',
+            value: 'outlined',
+          },
+        ],
+      },
+      label: {
+        type: 'textfield',
+        label: 'Label',
+        urlParam: 'label',
+        value: 'Name'
+      },
+      icons: {
+        type: 'filterchips',
+        name: 'Icons',
+        urlParam: 'icons',
+        value: iconsValue,
+        optionDescription: 'We recommend using Material Icons. ',
+        options: [{
+          label: 'Leading Icon',
+          value: 'leadingIcon'
+        }, {
+          label: 'Trailing Icon',
+          value: 'trailingIcon'
+        }]
+      }
+    },
+    order: [
+      'header', 'type', 'label', 'icons',
+    ],
+  };
+  return TextFieldConfig;
+}
+const TextFieldCatalog = (props) => (
   <ComponentCatalogPanel
     hero={<TextFieldHero />}
     title='Text Field'
-    description='Text fields allow users to input, edit, and select text. Text fields typically reside in forms but can appear in other places, like dialog boxes and search.'
+    description='Text fields allow users to input, edit, and select text.'
     designLink='https://material.io/go/design-text-fields'
     docsLink='https://material.io/components/web/catalog/input-controls/text-field/'
     sourceLink='https://github.com/material-components/material-components-web/tree/master/packages/mdc-textfield'
     demos={<TextFieldDemos/>}
+    initialConfig={getTextFieldConfig(props)}
+    {...props}
   />
 );
 
 this.clickEvent = (label) => () => ReactGA.event({category: gtagCategory, action: gtagTextFieldAction, label: label});
 
-export const TextFieldHero = () => {
+// Keep this until the material io experiment is complete.
+export const TextFieldHeroLegacy = () => {
```

Comment:
`TextFieldHero` and `ButtonHero` no longer need to be exported. They were exported for the iframe routes, but since we've switched to a `...Legacy` export so we don't disturb the experiment in progress.